fuel-vm/src/constraints/reg_key.rs

```diff
@@ -213,12 +215,12 @@ pub(crate) fn split_registers(registers: &mut [Word; VM_REGISTER_COUNT]) -> (Rea
         retl: RegMut(retl),
         flag: RegMut(flag),
     };
-    (r, WriteRegisters(rest))
+    (r, ProgramRegisters(rest))
 }
 
 pub(crate) fn copy_registers(
-    read_registers: &ReadRegistersRef<'_>,
-    write_registers: &WriteRegistersRef<'_>,
+    read_registers: &SystemRegistersRef<'_>,
+    write_registers: &ProgramRegistersRef<'_>,
```

Comment:
Nice, I love these name changes! Much clearer.

Reply:
Might also want to update the binding names too i.e. something like
```suggestion
    sys_regs: &SystemRegistersRef<'_>,
    prog_regs: &ProgramRegistersRef<'_>,
```